posts/inside-rust/2022-02-22-compiler-team-ambitions-2022.md

```diff
@@ -0,0 +1,424 @@
+
+If you want to help out with this, reach out to [jackh726] and [nikomatsakis].
+
+## Aspirations
+
+We would love help with any of the areas listed on this document, but this section specifically lists areas where we know we lack resources today.
+
+If you are interested in helping with any items here, please do reach out to the owner listed; they will be thrilled to talk to you.
+
+### P-high Aspirations (🦀)
+
+[pnkfelix] and [wesleywiser], as Compiler Team leads, are deploying processes to help us get a handle on the "high priority, but *not critical*" issues that the compiler has accumulated. We will be gradually identifying owners for each who will move progress forward, and in general working to keep better track of the set overall.
+
+If you would like to help with the task of reviewing or resolving such issues, reach out to [wesleywiser] and [apiraino], who are co-leads of WG-prioritization.
+
+### Debugging Aspirations (👩‍💻)
+
+We want better integration, at least with the popular debuggers. The command sequence to set up an idealized debugging experience is too obscure and therefore goes unused.
+
+We want to improve expression evaluation support: Today, most forms of method invocation do not work, because the debuggers do not know about Rust's method resolution rules.
+
+We want to revisit our debugger extension architecture for rendering Rust data structures, which is currently mostly independent sets of Python scripts.
+
+If you want to help out here, please reach out to [pnkfelix] and [wesleywiser].
+
+### Faster Builds Aspirations (👩‍💻, 🛠️)
+
+#### Parallel Compilation
+
+Parallel Compilation is one avenue for improving compiler performance. It is also a very complex area, especially when it comes to the tradeoff of how much of a hit one is willing to take on single core builds in order to enable more parallel computation. We already parallelize our LLVM invocations, but the parallelization of the rest of the compiler remains in an experimental state. This is an area we think needs long-term collaborative effort with the compiler team. We do not expect to deliver a solution here this year.
+
+If you want to discuss more with us about past attempts and ideas for the future, please reach out to [pnkfelix] and [wesleywiser].
+
+#### Incremental Compilation Aspirations
+
+Incremental compilation performance and stability are both ongoing concerns to the team. We *know* there is significant room to improve the effectiveness of incremental compilation, in terms of reducing the amount of redundant work done by successive `rustc` invocations.
+
+In addition, there is a significant amount of work that could be done to improve our testing infrastructure for incremental compilation which does not require deep knowledge of the compiler.
```

Comment:
```suggestion
In addition, there is a significant amount of work that could be done to improve our testing infrastructure for incremental compilation which does not require deep knowledge of the compiler. We have had to disable and subsequently reenable incremental compilation on the stable release; we want to expand our validation strategies so that we get alerted to problems in incremental compilation well before they come close to the stable channel.
```